frappe/core/doctype/submission_queue/submission_queue.py

```diff
@@ -205,3 +205,7 @@ def format_tb(traceback: str | None = None):
 		return
 
 	return traceback.strip().split("\n")[-1]
+
+
+def has_permission(doc, ptype=None, user=None):
+	return frappe.get_cached_doc(doc.ref_doctype, doc.ref_docname).has_permission()
```

Comment:
We shouldn't put things in the cache like this. Also, this doctype is already reading same doc multiple times.

Reply:
I'm also not sure about the cached thing ...maybe just simple get_doc would be fine?

Reply:
This has_perm thing can be skipped as well for now, in future when this feature is extended for queuing saves as well then this will be essential.